ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/queries/CommonQueries.scala

```diff
@@ -32,38 +30,28 @@ trait CommonQueries extends Queries {
   )(implicit connection: Connection): immutable.Seq[(Index, LedgerRecord)] =
     SQL"SELECT sequence_no, entry_id, envelope FROM #$LogTable WHERE sequence_no >= $start AND sequence_no < $end"
       .as(
-        (long("sequence_no") ~ byteArray("entry_id") ~ byteArray("envelope")).map {
+        (long("sequence_no") ~ binaryStream("entry_id") ~ byteArray("envelope")).map {
           case index ~ entryId ~ envelope =>
-            index -> LedgerRecord(
-              Offset(Array(index)),
-              DamlLogEntryId
-                .newBuilder()
-                .setEntryId(ByteString.copyFrom(entryId))
-                .build(),
-              envelope,
-            )
+            index -> LedgerRecord(Offset(Array(index)), DamlLogEntryId.parseFrom(entryId), envelope)
```

Comment:
I've made a note to change `LedgerRecord` to hide serialization aspects (i.e., to just ask for the bytes that have been written via `appendToLog` so you won't have to take care of parsing the log entry's key).